### PR TITLE
Open device and hex file through command line arguments

### PR DESCRIPTION
**Pull request by deerob4**
Building on top of Bence's recent PR in the launcher, this enables the Programmer to be launched with `--deviceSerial` and `--filePath` arguments that automatically open the given device and file.

### CODE REVIEW

File: src/reducers/argsReducer.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    name: 'args',
+    initialState,
+    reducers: {
+        argsParsed(state) {
+            state.argsParsed = true;
+        },
+    },
+    extraReducers: builder => {
+        builder.addDefaultCase((state, action) => {
+            console.log('DEILIG', action);
```
Comment (bencefr) [inline src/reducers/argsReducer.ts:29]: This looks very temporary...

**Review by bencefr — COMMENTED** [review]
Other than the console.log, it looks fine for me. I mean probably that whole `extraReducers` wasn't meant to be there.

**Review by bencefr — APPROVED** [review]

**Comment by datenreisender** [comment]
Besides some coding style differences from the rest of the code, I have mainly two concerns:
- Some of this code duplicates functionality that already exists elsewhere ([automatically selecting a device in `useAutoselectDevice.tsx` in shared](https://github.com/NordicSemiconductor/pc-nrfconnect-shared/blob/5f71a91a8449237e9540f522f025f132e1a4b1bc/src/Device/DeviceSelector/useAutoselectDevice.tsx).)
- It makes some deep assumptions about code in another project (shape of the state and existence/behaviour of actions) which I would like to avoid.

I think a better solution for this is not hard, but it is probably done easiest with also some changes in shared. I will code up something and then show it to you, so you can evaluate whether it also works for you.

**Comment by datenreisender** [comment]
@deerob4 Please have a look at #291, I think this simplifies the logic quite a bit (partially because some functionality was moved to shared, so you need the latest launcher for it to work). 

Because that PR partially reverts the changes from this PR, it may be easier to just look at what both PRs together look like: https://github.com/NordicSemiconductor/pc-nrfconnect-programmer/compare/master...command-args-cleanup

If you are happy with that other PR (and maybe have tested it), then feel free to merge that other PR into this one and the whole together into master.

**Comment by deerob4** [comment]
Thank you very much Marko, that works perfectly!

I didn't notice that there was already a `useAutoSelectDevice` hook in shared, and you're completely right that most of this belongs in there. It did feel icky to be intercepting the actions from shared, particularly when it was just the raw string constant.

**Pull request merged**